src/SFA.DAS.EmployerIncentives/Interfaces/ILegalEntitiesService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.InnerApi.Requests;
+using SFA.DAS.EmployerIncentives.Models;
+
+namespace SFA.DAS.EmployerIncentives.Interfaces
+{
+    public interface ILegalEntitiesService
+    {
+        Task<AccountLegalEntity[]> GetAccountLegalEntities(long accountId);
+        Task<AccountLegalEntity> GetLegalEntity(long accountId, long accountLegalEntityId);
+        Task DeleteAccountLegalEntity(long accountId, long accountLegalEntityId);
+        Task CreateLegalEntity(long accountId, AccountLegalEntityCreateRequest accountLegalEntity);
+        Task RefreshLegalEntities(IEnumerable<InnerApi.Responses.Accounts.AccountLegalEntity> accountLegalEntities, int pageNumber, int pageSize, int totalPages);
+        Task SignAgreement(long accountId, long accountLegalEntityId, SignAgreementRequest request);
```

Comment:
Can't the acocuntId/accountLegalEntityId go on the SignRequest?

Reply:
Done